litebird_sim/mpi.py

```diff
@@ -0,0 +1,44 @@
+# -*- encoding: utf-8 -*-
+
+try:
+    import mpi4py
+    from mpi4py import MPI
```

Comment:
Did you run it on a login node? This is a well-known limitation of NERSC login nodes, as explained here: https://docs.nersc.gov/programming/high-level-environments/python/mpi4py/

The fact that this error does not occur in Anaconda is not because this environment is «saner»; quite the contrary: it is unable to properly initialize the MPI system.